torch/_refs/__init__.py

```diff
@@ -1447,11 +1447,28 @@ def le(a: TensorLikeType, b: TensorLikeType) -> TensorLikeType:
 )
 def logaddexp(a: TensorLikeType, b: TensorLikeType) -> TensorLikeType:
     # Nb. this implementation does nto distribute the gradients evenly when a == b
-    mask = a >= b
+    mask = torch.real(a) >= torch.real(b)
     max_ = torch.where(mask, a, b)
     min_ = torch.where(mask, b, a)
-    inf_mask = torch.logical_and(torch.isinf(a), a == b)
-    return torch.where(inf_mask, a, max_ + torch.log1p(torch.exp(min_ - max_)))
+    inf_mask = torch.logical_and(
+        torch.logical_not(torch.isfinite(torch.real(a))), torch.real(a) == torch.real(b)
+    )
+    if utils.is_complex_dtype(a.dtype) or utils.is_complex_dtype(b.dtype):
+        # are you wondering what this bunch of codes are for? edge cases!
+        complex_tens = a if utils.is_complex_dtype(a.dtype) else b
+        neg_min_mask = torch.real(min_) < 0
+        inf_vals = torch.where(
+            neg_min_mask, min_, torch.log(torch.exp(min_) + torch.exp(max_))
+        )
+        non_nan_vals = torch.where(
+            inf_mask, inf_vals, max_ + torch.log1p(torch.exp(min_ - max_))
+        )
+        # the type for full_like does not include tensor yet
+        nan_vals = torch.full_like(complex_tens, complex(float("nan"), float("nan")))  # type: ignore[arg-type]
+        nan_mask = torch.isnan(min_)
+        return torch.where(nan_mask, nan_vals, non_nan_vals)
```

Comment:
No need to instantiate the full tensor, as broadcasting works.
```suggestion
        nan_mask = torch.isnan(min_)
        return torch.where(nan_mask, nan_vals, complex(float("nan"), float("nan")))
```